pandas/tests/tools/test_concat.py

```diff
@@ -1825,6 +1825,15 @@ def test_concat_bug_3602(self):
         result = concat([df1, df2], axis=1)
         assert_frame_equal(result, expected)
 
+    def test_concat_bug_15328(self):
+        df_empty = pd.DataFrame()
+        df_a = pd.DataFrame({'a': [1, 2]}, index=[0, 1])
+        result = pd.concat([df_empty, df_a], axis=1, join='inner')
+        self.assertTrue(result.empty)
+
+        result = pd.concat([df_a, df_empty], axis=1, join='inner')
+        self.assertTrue(result.empty)
+
     def test_concat_series_axis1_same_names_ignore_index(self):
```

Comment:
sorry i'm not clear on what you mean here. could you clarify our point to something similar ?